pythran/pythonic/types/numpy_expr.hpp

```diff
@@ -11,16 +11,29 @@ namespace pythonic
   namespace types
   {
 
+    namespace details
+    {
+      template <size_t value, class... Args>
+      array<long, value> init_shape(Args const &... args)
+      {
+        array<long, value> shape;
+        for (int i = 0; i < value; ++i) {
+          array<long, sizeof...(Args)> IShapes{
+              {(value <= Args::value ? args.shape()[i] : 0)...}};
```

Comment:
Ok, I understand, you compute max for each dimension to get the new shape but if value > Args::value (1D to 2D), you always have 0 for the 1D array while you should consider the shape of this array.

eg : a.shape -> (10,), b.shape -> (1, 3) => (a + b).shape ->(10, 3) and not (1, 3)

Am I wrong?